src/pages/studyView/table/StructuralVariantMultiSelectionTable.tsx

```diff
@@ -0,0 +1,782 @@
+              )
+            : this.props.promise.result || [];
+    }
+
+    @computed get flattenedFilters() {
+        return _.flatMap(this.props.filters);
+    }
+
+    @computed get selectableTableData() {
+        if (this.flattenedFilters.length === 0) {
+            return this.tableData;
+        }
+        return _.filter(
+            this.tableData,
+            data => !this.flattenedFilters.includes(data.uniqueKey)
+        );
+    }
+
+    @computed
+    get preSelectedRows() {
+        if (this.flattenedFilters.length === 0) {
+            return [];
+        }
+        const order = stringListToIndexSet(this.flattenedFilters);
+        return _.chain(this.tableData)
+            .filter(data => this.flattenedFilters.includes(data.uniqueKey))
+            .sortBy<StructVarMultiSelectionTableRow>(data =>
+                ifNotDefined(order[data.uniqueKey], Number.POSITIVE_INFINITY)
+            )
+            .value();
+    }
+
+    @computed
+    get preSelectedRowsKeys() {
+        return this.preSelectedRows.map(row => row.uniqueKey);
+    }
+
+    @computed
+    get tableColumns() {
+        return this.props.columns.map(column => {
+            const columnDefinition = this.getDefaultColumnDefinition(
+                column.columnKey,
+                this.columnsWidth[column.columnKey],
+                this.cellMargin[column.columnKey]
+            );
+            if (column.columnTooltip) {
+                columnDefinition.tooltip = column.columnTooltip;
+            }
+            return columnDefinition;
+        });
+    }
+
+    @action.bound
+    toggleModal(panelName: string) {
+        this.modalSettings.modalOpen = !this.modalSettings.modalOpen;
+        if (!this.modalSettings.modalOpen) {
+            return;
+        }
+        this.modalSettings.modalPanelName = panelName;
+    }
+
+    @action.bound
+    closeModal() {
+        this.modalSettings.modalOpen = !this.modalSettings.modalOpen;
+    }
+
+    @autobind
+    toggleCancerGeneFilter(event: any) {
+        event.stopPropagation();
+        this.props.onChangeCancerGeneFilter(!this.props.filterByCancerGenes);
+    }
+
+    @computed get isFilteredByCancerGeneList() {
+        return (
+            !!this.props.cancerGeneFilterEnabled &&
+            this.props.filterByCancerGenes
+        );
+    }
+
+    @computed get allSelectedRowsKeysSet() {
+        return stringListToSet([
+            ...this.props.selectedRowsKeys,
+            ...this.preSelectedRowsKeys,
+        ]);
+    }
+
+    @autobind
+    isChecked(uniqueKey: string) {
+        return !!this.allSelectedRowsKeysSet[uniqueKey];
+    }
+
+    @autobind
+    isDisabled(uniqueKey: string) {
+        return _.some(this.preSelectedRowsKeys, key => key === uniqueKey);
+    }
+
+    @action.bound
+    onStructVarHover(rowId: string, isHovered: boolean) {
+        if (isHovered) {
+            this.hoveredStructVarTableRowIds.push(rowId);
```

Comment:
in general i think it's a better practice to use immutable state.  i.e. don't mutate in place but update the whole array.  
```
this.hoveredStructVarTableRowIds = [rowId, ...this.hoveredStructVarTableRowIds]
```
Then you can make the array @observable.ref.   Ditto for the remove.